src/store/sled.rs

```diff
@@ -805,33 +804,43 @@ impl MessageStore for SledStore {
         }
     }
 
-    fn messages(&self, thread: &Thread, from: Option<u64>) -> Result<Self::MessagesIter, Error> {
+    fn messages(
+        &self,
+        thread: &Thread,
+        range: impl RangeBounds<u64>,
+    ) -> Result<Self::MessagesIter, Error> {
         let tree_thread = self.tree(self.messages_thread_tree_name(thread))?;
         debug!("{} messages in this tree", tree_thread.len());
-        let iter = if let Some(from) = from {
-            tree_thread.range(from.to_be_bytes()..)
-        } else {
-            tree_thread.range::<&[u8], std::ops::RangeFull>(..)
+
+        let iter = match (range.start_bound(), range.end_bound()) {
```

Comment:
That looks pretty ugly. Would be useful if [`Bound::map`](https://doc.rust-lang.org/nightly/core/ops/enum.Bound.html#method.map) was stable.

Reply:
`Bound::map` wouldn't help much here, since the `.range` function takes a `RangeBounds` as well.

Reply:
You could just use `(range.start_bound().map(|b| b.to_be_bytes()), range.end_bound().map(|b| b.to_be_bytes())` as bounds to the sled iterator if there would be a `map` I think. But, as mentioned, nothing you have to fix.

Reply:
Note: because `Bound::map` is not there yet (and there doesn't seem to be any news on that front in https://github.com/rust-lang/rust/issues/86026) you can see that `sled` does something similar https://docs.rs/sled/latest/src/sled/tree.rs.html#1207